### ISSUE TITLE
[IBUG] Can not turn off errors in prod.

### ISSUE THREAD

**Issue by JrIndie** [yiisoft/yii2#15232]
I have the following set:

error_reporting(E_ALL & ~E_NOTICE);
ini_set('display_errors', 0);
ini_set('log_errors', 1);
defined('YII_DEBUG') or define('YII_DEBUG', false);
defined('YII_ENV') or define('YII_ENV', 'prod');
defined('YII_ENABLE_ERROR_HANDLER') or define('YII_ENABLE_ERROR_HANDLER', false);
defined('YII_ENABLE_EXCEPTION_HANDLER') or define('YII_ENABLE_EXCEPTION_HANDLER', false);

Using Apache2, and yii2.

However, when I purposely try and cause an error by calling some unknown function, ill get the usual stack trace that I get inside dev mode.

**Comment by Alex-Code** [comment]
Where do you have that all set?

**Comment by JrIndie** [comment]
Inside a custom component that gets bootstrapped right after log,
**frontend/config/main.php**
`'bootstrap' => ['log', 'ConfigSettings'],`

**Comment by Alex-Code** [comment]
That'll probably be too late within the code.

The basic and advanced templates set the `YII_DEBUG` and `YII_ENV` constants in the `web/index.php` file so by the time your config file is loaded they're already defined.

You'll need to put them in the `index.php` file.

**Comment by JrIndie** [comment]
Okay, testing it now.  Is the web/index.php called every page?  Is there any other such files?  I was hoping to all my component before everything else to be able to set up configurations for production/dev modes.

**Comment by Alex-Code** [comment]
The `index.php` will be your entry point from the web that sets up the application.

Yii2 has `environments` where you can have configs for dev and production which can be switched using the `init` command.
https://github.com/yiisoft/yii2-app-advanced/blob/master/docs/guide/structure-environments.md

**Comment by JrIndie** [comment]
Okay thank you.  However, I get a white page now ...  Is there anything wrong with my settings?  ><

**Comment by Alex-Code** [comment]
Did it work before?

**Comment by JrIndie** [comment]
Yes, this is why my original post said I am moving into a production setting.

**Comment by JrIndie** [comment]
It seems like its halting loading the page.  ><

**Comment by JrIndie** [comment]
@Alex-Code  @samdark 

Huh?  Can you please advise me?  I spent the last year and a half developing using this software, and and my settings are correct, but it just shows a white page when in **prod**, in dev mode it will show the error log.  Obviously because I am doing an error on purpose so I can test to make sure it just ignores it.

Seems like the following is what is happening: ttp://www.yiiframework.com/forum/index.php/topic/22583-yii-debug-false-shows-php-notice-errors-and-stops/

**Comment by samdark** [comment]
Why disabling all the error handling?

**Comment by JrIndie** [comment]
Because in the event of a bug or some poor coding by some team members, I dont want anyone to see the stack trace, or any important information.  Especially  for script kiddies and php injection tricks to cause errors, I cant think of every single way to cause an error injecting php code to cause an error and prevent such things....

**Comment by samdark** [comment]
`defined('YII_DEBUG') or define('YII_DEBUG', false);` should be enough for the purpose.

**Comment by JrIndie** [comment]
Yes, and I am doing that, but it is just causing the page to be white and not respond (halting most likely).  I am inside a big project, so we have a lot of moving parts.  Maybe I will just start a new project and do my little test to make sure its working as intended and report back.

Thank you for your time.  You have an amazing product!

**Comment by JrIndie** [comment]
@samdark Sorry to bother you again.  Can you just let me know something before I head off?  I think I know what it doesnt work.  I was thinking Yii would be able to determine if a function does not exist and inspect it before it is called, and be able to handle it for me.  This does not seem to be the case, and I have been testing and doing this for days!

I have tested various type of errors, but all of them cause the page to halt (white). There are 13 total errors in PHP and I dont know how I can test all the other ones.

This was with your settings enabled: `defined('YII_DEBUG') or define('YII_DEBUG', false);`  And set in another page that is loaded.

```
Fatal Error();              // Halts Page
ParseError()                // Halts Page
include('WarningError');    // Halts Page
echo $NoticeError;          // Halts Page
```

So, it appears to me, that yii will always just halt the page.

Final Question:  Is there any errors in php that yii will find, and ignore and gracefully act as if nothing as happened?

Thank you so much!

**Comment by samdark** [comment]
> Final Question: Is there any errors in php that yii will find, and ignore and gracefully act as if nothing has happened, besides the above two?

No. The two above should be handled as well if you're not turning off error handlers.

**Comment by samdark** [comment]
Would you please provide steps to reproduce the issue starting with basic application template?

**Comment by Renkas** [comment]
@JrIndie have a look at PHP's error log if you are getting a white page.

**Comment by terabytesoftw** [comment]
Thanks for the report.

I rechecked this against the current Yii2 master (May 2026) and could not reproduce a framework bug. The expected production setup is to define `YII_DEBUG=false` and `YII_ENV='prod'` at the very beginning of the entry script, and leave Yii error handler enabled. With that setup, Yii renders the generic 500 error view and does not expose the stack trace.

Defining these constants from a bootstrapped component is too late, because the application registers the error handler before bootstrap components run. Also, `YII_ENABLE_EXCEPTION_HANDLER` is not a Yii2 constant, and setting `YII_ENABLE_ERROR_HANDLER=false` disables Yii error handling entirely, so PHP/server configuration controls the output.

Uncaught fatal errors cannot be safely ignored or resumed. If this still happens in a supported Yii2 version, please provide a minimal reproduction based on the basic application template.

Closing as not reproducible.
